github/codeowners.py

```diff
@@ -210,3 +224,65 @@ def resolve_email_addresses(
                 github_api=github_api,
             )
             continue
+
+
+def resolve_usernames(
+    codeowners_entries: typing.Iterable[Username | EmailAddress | Team],
+    github_api: GitHub,
+    deduplicate: bool=False,
```

Comment:
thinking of it: is there any case where we would not want to deduplicate? I cannot think of any
-> just always deduplicate; store all users in a set, and leverage its implicit deduplication